lib/internal/bridge/bluetooth_state_mixin.dart

```diff
@@ -14,4 +17,28 @@ mixin BluetoothStateMixin on FlutterBLE {
       ArgumentName.transactionId: transactionId,
     });
   }
+
+  Future<BluetoothState> state() => _methodChannel
+      .invokeMethod(MethodName.getState)
+      .then(_mapToBluetoothState);
+
+  Stream<BluetoothState> onStateChange() async* {
+    await _methodChannel.invokeMethod(MethodName.observeState);
+    yield* _adapterStateChanges
+        .receiveBroadcastStream()
+        .map(_mapToBluetoothState);
+  }
+
+  BluetoothState _mapToBluetoothState(dynamic rawValue) {
+    switch (rawValue) {
+      case "PoweredOn":
+        return BluetoothState.POWERED_ON;
```

Comment:
Aren't there also `TURNING_ON`, `TURNING_OFF`?

Reply:
https://developer.apple.com/documentation/corebluetooth/cbmanagerstate

There are, but not on iOS and we need to conform

Reply:
Yes there are, abut apparently iOS doesn't support them so we introduced resetting state. I also added unknown, unsupported and unauthorized states as it is in RN lib.